### PR TITLE
[WASMEDGE-Zlib] Working example with Build + Run Guide.

### PR DESCRIPTION
**Pull request by notfathomless**
Companion example for the zlib plugin for wasmedge created by PR #2562.

To whom it may concern.
@hydai

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: examples/plugin/wasmedge-zlib/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# WasmEdge WASMEDGE-Zlib example.
+
+This is a example to demonstrate how to use wasmedge-zlib plugin of WasmEdge with C++.
```
Comment (hydai) [inline examples/plugin/wasmedge-zlib/README.md:3]: Please use tools to fix several grammar issues. E.g. grammarly.

File: examples/plugin/wasmedge-zlib/main.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+#define Z_FINISH 4
+#define Z_NULL NULL
+
+typedef void *(*alloc_func)(void *opaque, unsigned int items,
+                            unsigned int size);
+typedef void (*free_func)(void *opaque, void *address);
+
+struct internal_state;
+
+typedef struct z_stream_s {
```
Comment (hydai) [inline examples/plugin/wasmedge-zlib/main.cpp:40]: Is there any reason not to apply the coding style to this structure?

**Review by hydai — CHANGES_REQUESTED** [review]

File: examples/plugin/wasmedge-zlib/README.md (hunk 1 above)
Comment (notfathomless) [reply examples/plugin/wasmedge-zlib/README.md]: Done.

File: examples/plugin/wasmedge-zlib/main.cpp (hunk 2 above)
Comment (notfathomless) [reply examples/plugin/wasmedge-zlib/main.cpp:40]: @hydai 
Ideally the user should not be writing these zlib internal structures.

I think the ideal way here will be to provide a `.wit` file to describe all the zlib structures & all.
Then use [wit-bindgen](https://github.com/bytecodealliance/wit-bindgen) to create these structures on the fly.

I created this PR, to provide an example to the `wasmedge-zlib` for the time being.


So the usage of `WASM Components` should be the ideal

**Review by hydai — APPROVED** [review]

**Comment by hydai** [comment]
Thanks!

**Pull request merged**